components/script/dom/globalscope.rs

```diff
@@ -202,6 +223,30 @@ impl GlobalScope {
         GlobalScope::from_object(obj)
     }
 
+    pub fn add_uncaught_rejection(&self, rejection: HandleObject) {
+        self.uncaught_rejections.borrow_mut().push(Heap::boxed(rejection.get()));
+    }
+
+    pub fn remove_uncaught_rejection(&self, rejection: HandleObject) {
+        let mut uncaught_rejections = self.uncaught_rejections.borrow_mut();
+
+        if let Some(index) = uncaught_rejections.iter().position(|promise| *promise == Heap::boxed(rejection.get())) {
+            uncaught_rejections.remove(index);
+        }
+    }
+
+    pub fn get_uncaught_rejections(&self) -> &DomRefCell<Vec<Box<Heap<*mut JSObject>>>> {
+        &self.uncaught_rejections
+    }
+
+    pub fn add_consumed_rejection(&self, rejection: HandleObject) {
+        self.consumed_rejections.borrow_mut().push(Heap::boxed(rejection.get()));
+    }
+
+    pub fn get_consumed_rejections(&self) -> &DomRefCell<Vec<Box<Heap<*mut JSObject>>>> {
```

Comment:
Nothing ever calls `get_consumed_rejections`, so the list just grows over time. Let's remove the list if we're not going to use it.